views-core/src/main/java/io/micronaut/views/model/MapViewModelProcessor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.views.model;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.views.ModelAndView;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Abstract class to ease populating a map model and handle immutable maps.
+ */
+@Internal
+public abstract class MapViewModelProcessor implements ViewModelProcessor<Map<String, Object>> {
+    @Override
+    public void process(@NonNull HttpRequest<?> request, @NonNull ModelAndView<Map<String, Object>> modelAndView) {
+        Map<String, Object> viewModel = modelAndView.getModel().orElseGet(() -> {
+            final HashMap<String, Object> newModel = new HashMap<>(1);
+            modelAndView.setModel(newModel);
+            return newModel;
+        });
+        try {
+            populateModel(request, viewModel);
+        } catch (UnsupportedOperationException ex) {
+            final HashMap<String, Object> modifiableModel = new HashMap<>(viewModel);
```

Comment:
yes `Map` on the left hand side

Reply:
resolved via https://github.com/micronaut-projects/micronaut-views/pull/914/commits/d15e09247b6832a1b98f8c0af2e81edd215fbb73